pkg/block/fetcher.go

```diff
@@ -35,32 +35,35 @@ import (
 	"github.com/thanos-io/thanos/pkg/runutil"
 )
 
-type fetcherMetrics struct {
-	syncs        prometheus.Counter
-	syncFailures prometheus.Counter
-	syncDuration prometheus.Histogram
+// FetcherMetrics holds metrics tracked by the metadata fetcher. This struct and its fields are exported
+// to allow depending projects (eg. Cortex) to implement their own custom metadata fetcher while tracking
+// compatible metrics.
+type FetcherMetrics struct {
```

Comment:
What's wrong with this approach? 🤔 

Potentially: Fetcher hasn't been designed to be replaced with a different implementation. 

I think this the evolution of package that needs more implementations than one.